src/main.ts

```diff
@@ -13,24 +13,61 @@ export const URI = 'Dataway';
 
 export type URI = typeof URI;
 
+/**
+ * Interface usually used as default value for dataway,
+ * contains no data
+ */
 export interface NotAsked {
   readonly _tag: 'NotAsked';
 }
 
+/**
+ * Interface used to represent a remote datasource being loaded or
+ * a command being processed (ex: form submit)
+ */
 export interface Loading {
   readonly _tag: 'Loading';
 }
-
+/**
+ * Interface used to represent any failure requesting a remote datasource
+ * It can be used to
+ * 1. represent and store an http error
+ * 2. represent and store a value decoding/validation error
+ * 3. a computation error on the retrieved data
```

Comment:
```suggestion
 * 3. represent and store a computation error on the retrieved data
```
sinon la phrase ne fonctionne pas. Une autre solution est de ne mettre "represent and store" qu'une seule fois, avant l'énumération.

Reply:
oui en effet top